grails-app/views/product/edit.gsp

```diff
@@ -373,7 +373,7 @@
                             <g:render template="productPackages" model="[productInstance:productInstance]"/>
 						</div>
                         <div id="inventory-level-dialog" class="dialog hidden" title="Add a new stock level">
-                            <g:render template="../inventoryLevel/form" model="[productInstance:productInstance,inventoryLevelInstance:new InventoryLevel()]"/>
+                            <g:render template="../inventoryLevel/form" model="[productInstance:productInstance,inventoryLevelInstance:new InventoryLevel(),currentLocation:session.warehouse]"/>
```

Comment:
Again, if you reference `currentLocation?.inventory` in `_form.gsp` you'll get a lazy initialization exception.